### ISSUE TITLE
External object store can't reference a "parent" object store in another namespace

### ISSUE THREAD

**Issue by BlaineEXE** [rook/rook#10489]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

A CephObjectStore (ns: rook-ceph-secondary) cannot specify (via `externalRgwEndpoints`) the Service IP (ClusterIP) of a CephObjectStore's RGW cluster in another Kubernetes namespace (ns: rook-ceph). 

Rook creates a Service for the external CephObjectStore (ns: rook-ceph-secondary) that points to all of the `externalRgwEndpoints`, and this Service is used by the admin API to do operations on the parent CephObjectStore. However, Kubernetes Services can't seem to route to other Services (this is captured in a stackoverflow question I found below).

https://stackoverflow.com/questions/37221483/service-located-in-another-namespace

This results in the external CephObjectStore not being able to connect. Its status is below:
```
Status:
  Bucket Status:
    Details:       failed to get details from ceph object user "rook-ceph-internal-s3-user-checker-7021944e-2810-4694-8f85-cc8fedca8b52": Get "http://rook-ceph-rgw-external-store.rook-ceph-secondary.svc:80/admin/user?format=json&uid=rook-ceph-internal-s3-user-checker-7021944e-2810-4694-8f85-cc8fedca8b52": dial tcp 10.96.84.229:80: i/o timeout
    Health:        Failure
    Last Changed:  2022-06-22T07:42:00Z
    Last Checked:  2022-06-22T17:10:29Z
  Info:
    Endpoint:           http://rook-ceph-rgw-external-store.rook-ceph-secondary.svc:80
  Observed Generation:  6
  Phase:                Failure
```

From the `rook-ceph-operator` pod, I am able to curl the parent RGW's (ns: rook-ceph) service IP directly.
```
$ curl 10.98.111.221:80
<?xml version="1.0" encoding="UTF-8"?><ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Owner><ID>anonymous</ID><DisplayName></DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>
```

However, I am not able to curl the external RGW's service (ns: rook-ceph-secondary) service IP. It times out
```
$ curl 10.96.84.229:80
curl: (7) Failed to connect to 10.96.84.229 port 80: Connection timed out
```

The k8s service in rook-ceph-secondary is configured such that it should be pointing correctly to the rook-ceph service (it has IP 10.98.111.221).
```
Name:              rook-ceph-rgw-external-store
Namespace:         rook-ceph-secondary
Labels:            app=rook-ceph-rgw
                   app.kubernetes.io/component=cephobjectstores.ceph.rook.io
                   app.kubernetes.io/created-by=rook-ceph-operator
                   app.kubernetes.io/instance=external-store
                   app.kubernetes.io/managed-by=rook-ceph-operator
                   app.kubernetes.io/name=ceph-rgw
                   app.kubernetes.io/part-of=external-store
                   ceph_daemon_id=external-store
                   ceph_daemon_type=rgw
                   rgw=external-store
                   rook.io/operator-namespace=rook-ceph
                   rook_cluster=rook-ceph-secondary
                   rook_object_store=external-store
Annotations:       <none>
Selector:          <none>
Type:              ClusterIP
IP Family Policy:  SingleStack
IP Families:       IPv4
IP:                10.96.84.229
IPs:               10.96.84.229
Port:              http  80/TCP
TargetPort:        80/TCP
Endpoints:         10.98.111.221:80
Session Affinity:  None
Events:            <none>
```

**How to reproduce it (minimal and precise):**

I am testing with these 4 manifests on a standard minikube cluster.
- `cluster-test.yaml`: the "parent" test cluster i(n the rook-ceph namespace)
- `object-multisite-test.yaml`: the "parent" object store (with multisite CRDs also)
- `cluster-secondary-test.yaml`: the external cluster (in the rook-ceph-secondary namespace)
- `object-secondary-external-test.yaml`: the external object store which references the RGW created in `object-multisite-test.yaml`

[manifests.zip](https://github.com/rook/rook/files/8959918/manifests.zip)

**Discussion**
This is getting in the way of me easily testing external mode CephObjectStores; however, I don't know that this is strictly a bug. I think this is the point we have to discuss. 

I think there are 2 basic options:
1. Rook could take the stance that we do not support "external" RGWs merely being RGWs from a different Kubernetes namespace. This would require no fix other than to document that we don't support that.
2. There are reasons we might support it. We support this for CephClusters, so supporting it for object follows the same trend. It would make testing the external mode object stores in CI easier as well. Finally, this could be a use-case that some users may want.

Regarding option 2: I haven't explored all possible solutions. There are some other things I can explore that don't involve code changes.
-I may be able to specify the `v1.EndpointAddress` differently. IP is required, but `ObjectReference` and `hostname` are other options.
- It may be that we really just require users to create a special kind of Service for the "parent" object store (ns: rook-ceph in my example) to be the reference, and that should be straightforward.

**Comment by BlaineEXE** [comment]
@parth-gr @travisn @leseb ^

**Comment by mgugino-uipath** [comment]
Are you sure you don't have a network policy preventing communication or something?  This works for me using the answer accepted on SO:

```
#svc
kind: Service
apiVersion: v1
metadata:
  name: rook-ceph-rgw
  namespace: default
spec:
  type: ExternalName
  externalName: rook-ceph-rgw-rook-ceph.rook-ceph.svc.cluster.local
  ports:
  - port: 80
```

```
[root@my-shell /]# curl http://rook-ceph-rgw
<?xml version="1.0" encoding="UTF-8"?><ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Owner><ID>anonymous</ID><DisplayName></DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>[root@my-shell /]#
[root@my-shell /]# curl http://rook-ceph-rgw-rook-ceph.rook-ceph.svc.cluster.local
<?xml version="1.0" encoding="UTF-8"?><ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Owner><ID>anonymous</ID><DisplayName></DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>
```

**Comment by thotz** [comment]
yaa exposing internal rgw service via Nodeport so that IP of the host will be accessible from other namespace. I can main issue here is that `Endpoint` struct for `Objectstorespec` only supports IP than fqdn

**Comment by BlaineEXE** [comment]
> yaa exposing internal rgw service via Nodeport so that IP of the host will be accessible from other namespace. I can main issue here is that `Endpoint` struct for `Objectstorespec` only supports IP than fqdn

This is correct. With the current Rook code/behavior, I can't think of a way a user can create a Service that would make it so the health checker routine could validate the external connection to the parent RGW.

**Comment by BlaineEXE** [comment]
Resolved by #11208
